compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -873,7 +873,8 @@ impl<'a> Parser<'a> {
             id: DUMMY_NODE_ID,
             value: self.mk_expr(blk.span, ExprKind::Block(blk, None), AttrVec::new()),
         };
-        Ok(self.mk_expr(span, ExprKind::ConstBlock(anon_const), AttrVec::new()))
+        let blk_span = anon_const.value.span;
+        Ok(self.mk_expr(span.to(blk_span), ExprKind::ConstBlock(anon_const), AttrVec::new()))
```

Comment:
isn't what you're saying `span.to(blk.span)`?

Reply:
That's the idea yes, but I feel like a local will be needed since `blk` gets moved in the `mk_expr` call above to create the `AnonConst`. I suppose I could move the local before it is moved if you'd prefer `blk_span = blk.span`

Reply:
Makes sense :+1: